External resources: Add repro "gmt-pygmt-plotting"

yvonnefroehlich opened the pull request:
**Description of proposed changes**

This PR adds the repository [gmt-pygmt-plotting](https://github.com/yvonnefroehlich/gmt-pygmt-plotting) to the external resources page. (Some folders are missing. These Jupyter notebooks or Python scripts will be uploaded up on acceptance of the related publication.)

Currently, I am not 100% sure about the link to the image. For the other entries, the links are significantly shorter and of the form `https://user-images.githubusercontent.com/*` instead of `https://private-user-images.githubusercontent.com/*`. 

**Preview**:

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.
- [ ] Use underscores (not hyphens) in names of Python files and directories.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash command is:

- `/format`: automatically format and lint the code

seisman commented:
it's likely because the image was uploaded in a private repository. maybe try drag the image to the comment box of this PR.

yvonnefroehlich commented:
> it's likely because the image was uploaded in a private repository. maybe try drag the image to the comment box of this PR.

Thanks! I tried this: 

![github_maps_readme_main](https://github.com/GenericMappingTools/pygmt/assets/94163266/21a91be1-b14e-4809-b7e7-d6e4dc94587d)

But it is still a link in the form of `https://private-user-images.githubusercontent.com/*` created. Anyway, for now, I created a folder with all the images for the READMEs and added this link. This seems to work 🙂.

michaelgrund commented:
I adjusted the PR title, it's called repo (for repository) not repro.

yvonnefroehlich commented:
> I adjusted the PR title, it's called repo (for repository) not repro.

Thanks for spotting the typo 🙂!
Actually I changed it now to "repository" as is  sounds / looks better in the Changelog.
